src/components/item/settings/ItemSettingsButton.tsx

```diff
@@ -15,22 +18,48 @@ import { BUILDER } from '../../../langs/constants';
 
 type Props = {
   id: string;
+  type?: ActionButtonVariant;
+  title?: string;
 };
 
-const ItemSettingsButton = ({ id }: Props): JSX.Element => {
+const ItemSettingsButton = ({
+  id,
+  type = ActionButton.ICON_BUTTON,
+  title = 'Settings',
+}: Props): JSX.Element => {
   const { t: translateBuilder } = useBuilderTranslation();
-  return (
-    <Tooltip title={translateBuilder(BUILDER.SETTINGS_TITLE)}>
-      <IconButton
-        component={Link}
-        to={buildItemSettingsPath(id)}
-        className={ITEM_SETTINGS_BUTTON_CLASS}
-        id={buildSettingsButtonId(id)}
-      >
-        <SettingsIcon />
-      </IconButton>
-    </Tooltip>
-  );
+
+  switch (type) {
+    case ActionButton.MENU_ITEM:
+      return (
+        <MenuItem
+          key={title}
+          className={ITEM_SETTINGS_BUTTON_CLASS}
+          component={Link}
+          id={buildSettingsButtonId(id)}
+          to={buildItemSettingsPath(id)}
+        >
+          <ListItemIcon>
+            <SettingsIcon />
+          </ListItemIcon>
+          {title}
+        </MenuItem>
+      );
+    case ActionButton.ICON_BUTTON:
+    default:
+      return (
+        <Tooltip title={translateBuilder(BUILDER.SETTINGS_TITLE)}>
```

Comment:
Either use the `title` prop or do not offer a prop to set the title. 
if you decide to use the prop do not make it optional.